code/datums/storage/subtypes/rped.dm

```diff
@@ -9,7 +9,77 @@
 	max_specific_storage = WEIGHT_CLASS_NORMAL
 	numerical_stacking = TRUE
 
+	var/list/allowed_mat_types=list(
+		/obj/item/stack/sheet/glass,
+		/obj/item/stack/sheet/plasteel,
+		/obj/item/stack/cable_coil
+	)
+
+	var/list/allowed_bluespace_types=list(
+		/obj/item/stack/ore/bluespace_crystal,
+		/obj/item/stack/ore/bluespace_crystal/refined,
+		/obj/item/stack/ore/bluespace_crystal/artificial,
+		/obj/item/stack/sheet/bluespace_crystal
+	)
+
+
 /datum/storage/rped/can_insert(obj/item/to_insert, mob/user, messages = TRUE, force = FALSE)
 	. = ..()
-	if(!to_insert.get_part_rating())
+
+	///we check how much of glass,plasteel & cable the user can insert
+	if(isstack(to_insert))
+		///user tried to insert invalid stacktype
+		if(!is_type_in_list(to_insert,allowed_mat_types) && !is_type_in_list(to_insert,allowed_bluespace_types))
+			return FALSE
+
+		var/obj/item/stack/the_stack=to_insert
+		//how much of the stack is the user trying to insert
+		var/insert_amount=the_stack.amount
+		//how much of this stack type is currently in the stack
+		var/present_amount=0
+		//what is the maximum allowed amount for this stack type in storage
+		var/max_amount=30
```

Comment:
Did you mean to have these declared on the typepath? Right now these only exist here and cannot be modified.